### PR TITLE
Added compilation support for Node.js v4

### PR DESCRIPTION
**Pull request by braydonf**
Closes: https://github.com/bitpay/bitcore/issues/1361
Closes: https://github.com/bitpay/bitcore-node/issues/259

### CODE REVIEW

File: etc/bitcoin.patch [hunk 1]
```diff
@@ -268,7 +292,7 @@ index cce687a..0f162ff 100644
  {
      SetupEnvironment();
 @@ -175,3 +180,4 @@ int main(int argc, char* argv[])
-
+
```
Comment (kleetus) [inline etc/bitcoin.patch]: is there a stray space here? I can't tell exactly.

Comment (braydonf) [inline etc/bitcoin.patch]: I'll remove it, there is a space there.

**Comment by kleetus** [comment]
this worked ok on debian jesse, but on Mac 10.10, it get:

module.js:339
    throw err;
    ^

Error: Cannot find module 'db/package.json'
    at Function.Module._resolveFilename (module.js:337:15)
    at Function.Module._load (module.js:287:25)
    at Module.require (module.js:366:17)
    at require (module.js:385:17)
    at Function.setupServices (/Users/chrisk/source/zzbitcore_node/lib/scaffold/start.js:51:30)
    at start (/Users/chrisk/source/zzbitcore_node/lib/scaffold/start.js:203:31)
    at Object.<anonymous> (/Users/chrisk/source/zzbitcore_node/bin/start.js:8:1)
    at Module._compile (module.js:435:26)
    at Object.Module._extensions..js (module.js:442:10)
    at Module.load (module.js:356:32)

when running:

node bin/start

node 4.2.1

**Comment by braydonf** [comment]
I don't think `bin/start` is supported anymore.

**Comment by kleetus** [comment]
works using ./bin/bitcore start and removing insight related plugins from bitcore-node.json

**Comment by kleetus** [comment]
LGTM

**Comment by pnagurny** [comment]
tested - works locally. LGTM

**Pull request merged**